validate/validate.go

```diff
@@ -335,6 +336,24 @@ func (v *Validator) CheckMounts() (msgs []string) {
 	return
 }
 
+func (v *Validator) CheckOS() (msgs []string) {
+	logrus.Debugf("check os")
+
+	if v.spec.Platform.OS != "linux" {
+		if v.spec.Linux != nil {
+			msgs = append(msgs, fmt.Sprintf("When system is not linux, Linux field should be empty"))
```

Comment:
> But spec has changed in [this][1] PR.

Ah, I'd missed that (that PR changed a lot ;).  I'm not sure how the new approach will scale to a [`vm` section][2] if/when VM support lands in the spec, but I guess we'll see.

Anyhow, thanks for re-citing master.  I'm fine with your implementation landing as soon as runtime-tools switches from targeting v1.0.0-rc4 (which has the old language) to rc5 (or whichever spec release we target next which has the new language).

As a minor nit, I'd recommend wording that more closely follows the spec.  Something like:

    msgs = append(msgs, fmt.Sprintf("'linux' MUST NOT be set when platform.os is %q", v.spec.Platform.OS))

[1]: https://github.com/opencontainers/runtime-spec/pull/673
[2]: https://github.com/opencontainers/runtime-spec/pull/405#issuecomment-263702840